dusk/grammar.py

```diff
@@ -184,13 +193,11 @@ def type(self, node):
             ctx=Load,
         )
     )
-    def field_type(self, hindex: expr = None, vindex: str = None):
-
-        if hindex is None:
-            return make_field_dimensions_vertical()
-
-        return make_field_dimensions_unstructured(
-            self.location_chain(hindex), 1 if vindex is not None else 0
+    def field_type(self, field_type: str, hindex: expr = None, vindex: str = None):
+        return (
+            field_type,
+            self.location_chain(hindex) if hindex is not None else None,
+            1 if vindex is not None else 0,
```

Comment:
I realize (now) that 1 is a mask. A questionable choice in dawn and not dusks' problem in the least, but maybe we could somehow make more explicit that this is a mask in the code? even if its just using a constant kmask=1 instead of a "raw" 1?

Reply:
Fair point, but I think in this case it's hard to do that. The tuple that's returned here is consumed on line 163 (of the new changes). We could use a dict, but python doesn't have destructuring for dicts. The whole thing is a bit messy, since we added a second type. I'd like to have proper typing support in dusk, then we could do this way better.